server/controllers/match/placementRequests/spaceBookingsController.test.ts

```diff
@@ -1,98 +1,103 @@
 import type { NextFunction, Request, Response } from 'express'
 import { DeepMocked, createMock } from '@golevelup/ts-jest'
 
-import { faker } from '@faker-js/faker'
+import { when } from 'jest-when'
 import { Cas1SpaceBookingCharacteristic } from '@approved-premises/api'
 import SpaceBookingsController from './spaceBookingsController'
 
-import { PlacementRequestService, SpaceService } from '../../../services'
+import { PlacementRequestService, PremisesService, SpaceService } from '../../../services'
 import {
+  cas1PremisesSummaryFactory,
   cas1SpaceBookingFactory,
   newSpaceBookingFactory,
-  personFactory,
   placementRequestDetailFactory,
   spaceBookingRequirementsFactory,
 } from '../../../testutils/factories'
-import { filterOutAPTypes, occupancyViewLink, placementDates } from '../../../utils/match'
 import paths from '../../../paths/admin'
-import { fetchErrorsAndUserInput } from '../../../utils/validation'
-import { occupancyCriteriaMap } from '../../../utils/match/occupancy'
+import matchPaths from '../../../paths/match'
+import * as validationUtils from '../../../utils/validation'
+import { createQueryString } from '../../../utils/utils'
+import { spaceBookingConfirmationSummaryListRows } from '../../../utils/match'
 
-jest.mock('../../../utils/validation')
 describe('SpaceBookingsController', () => {
   const token = 'SOME_TOKEN'
 
-  const request: DeepMocked<Request> = createMock<Request>({ user: { token } })
+  let request: Readonly<DeepMocked<Request>>
   const response: DeepMocked<Response> = createMock<Response>({})
   const next: DeepMocked<NextFunction> = createMock<NextFunction>({})
 
   const placementRequestService = createMock<PlacementRequestService>({})
+  const premisesService = createMock<PremisesService>({})
   const spaceService = createMock<SpaceService>({})
 
+  const premises = cas1PremisesSummaryFactory.build()
+  const placementRequestDetail = placementRequestDetailFactory.build({ duration: 84 })
+
   let spaceBookingsController: SpaceBookingsController
 
   beforeEach(() => {
     jest.resetAllMocks()
-    spaceBookingsController = new SpaceBookingsController(placementRequestService, spaceService)
+    spaceBookingsController = new SpaceBookingsController(placementRequestService, premisesService, spaceService)
+    request = createMock<Request>({ user: { token } })
+
+    placementRequestService.getPlacementRequest.mockResolvedValue(placementRequestDetail)
+    premisesService.find.mockResolvedValue(premises)
+
+    jest.spyOn(validationUtils, 'fetchErrorsAndUserInput')
+    when(validationUtils.fetchErrorsAndUserInput as jest.Mock)
```

Comment:
Not sure if I'm missing something here. Isn't the idea of jest-when to allow a mocked return based on the received values?
As this there is only one when, and it returns a constant - I don't see the difference from just using `<mock>.mockReturnValue()`

Reply:
We shouldn't actually need to mock `fetchErrorsAndUserInput` -- we can simply provide a mocked `flash()` method to the existing request mock, which would then return the default values for the above.